tests/Integration/Behaviour/Features/Scenario/Discount/BO/add_discount_with_code.feature

```diff
@@ -0,0 +1,102 @@
+# ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml -s discount --tags add-discount-with-code
```

Comment:
Perfect for the BO side, now we just need a new feature on the FO side and to test one discount that doesn't rely on a promo code but is automatic instead and check that it is applied automatically

One scenario is enough for now, maybe you can rely on the minimum product quantity as it's easy to setup